keps/sig-network/3698-multi-network-requirements/README.md

```diff
@@ -0,0 +1,375 @@
+  used for the default gateway.
+
+### User Stories
+
+All user stories represent the type of use cases the multi-networking API should
+be able to support. References to technologies or exact products does not
+indicate that this API will directly support them. The set of requirements,
+defined out of these use cases, are the final indicator of what will be covered
+by this API and effort.
+
+#### Story #1
+As a Cloud Native Network Function (CNF) vendor I require an additional
+interface to be provisioned into the Kubernetes Pod. Each of these interfaces
+has to be in an isolated network for regulatory compliance. The isolation has to
+be done on a Layer-2.
+
+<p align="center">
+  <img src="mn-story-1.png?raw=true" alt="multi-network story 1 network L2 isolation"/>
+</p>
+
+#### Story #1a
+As a Cloud NativeNetwork Function (CNF) vendor I require an additional interface
+to be provisioned into the Kubernetes Pod. Each of these interfaces has to be in
+an isolated network for regulatory compliance. The isolation has to be done on a
+Layer-3.
+
+<p align="center">
+  <img src="mn-story-1a.png?raw=true" alt="multi-network story 1a network L3 isolation"/>
+</p>
+
+#### Story #2
+As a Cloud Native Network Function (CNF) vendor I require a HW-based interface
+(e.g. SRIOV VF) to be provisioned to my workload Pod. I need to leverage that HW
+for performance purposes (high bandwidth, low latency), that my user-space
+application (e.g. DPDK-based) can use. The VF will not use the standard netdev
+kernel module. The Pod’s scheduling to the nodes should be based on hardware
+availability (e.g. devicePlugin or some other way).
+
+<p align="center">
+  <img src="mn-story-2.png?raw=true" alt="multi-network story 2 HW interface"/>
+</p>
+
+#### Story #3
+I have implemented my Kubernetes cluster networking using a virtual switch. In
+this implementation I am capable of creating isolated Networks. I need a means
+to express to which Network my workloads connect to.
+
+<p align="center">
+  <img src="mn-story-3.png?raw=true" alt="multi-network story 3 virtual switch isolation"/>
+</p>
+
+#### Story #4
+As a Virtual Machine -based compute platform provider that I run on top of
+Kubernetes and Kubevirt I require multi-tenancy. The isolation has to be
+achieved on Layer-2 for security reasons.
+
+<p align="center">
+  <img src="mn-story-4.png?raw=true" alt="multi-network story 4 Kubevirt VMs"/>
+</p>
+
+#### Story #5
+As a platform operator I need to connect my on-premise networks to my workload
+Pods. I need to have the ability to represent these networks in my Kubernetes
+cluster in such a way that I can easily use them in my workloads.
+
+<p align="center">
+  <img src="mn-story-5.png?raw=true" alt="multi-network story 5 On-Premise Network representation"/>
+</p>
+
+#### Story #6
+As a Kubernetes cluster administrator I wish to isolate workloads based on
+namespaces and network access via assigning different default Network to a
```

Comment:
Could you please clarify 'default Network' in the story more because the term 'default network' seems to be a bit different from 'default network' in 'Terminology' section.

In the 'Terminology' section, default network is the initial cluster-wide Pod networking, hence the term, 'different default Network', seems to be weird to me.

Reply:
Yeah, here it is about the "is available to the Pod when no additional networking configuration is provided" part of that definition so it is partially right, but incorrect in the part you mention. Should we just call this "namespace-default" Network?
"namespace-default" Network - network to which Pod is connected to when no networking configuration is specified in Pod, currently it is always the cluster default Network

Reply:
I agree that we should define new term in 'Terminology'.

> "namespace-default" Network - network to which Pod is connected to when no networking configuration is specified in Pod, currently it is always the cluster default Network

Have another question. Based on your description, 'namespace-default' Network is added if 'no network configuration'. Does this mean this network is only added if no network is configured? (I mean that this network is not added when some network is configured)?

In addition, let me clarify that 'namespace-default' network is unique in each namespace or not (to clarify why you add 'namespace' in the term).

Reply:
> Does this mean this network is only added if no network is configured? (I mean that this network is not added when some network is configured)?

That is correct, there is no assumption, only explicit list of what to attach to, see req 10 below.

> In addition, let me clarify that 'namespace-default' network is unique in each namespace or not (to clarify why you add 'namespace' in the term).

This story is about adding such capability, having different default per namespace. Initially there will be only 1 cluster-wide default. This story is in phase IV, so probably we would return to this in next year or so.